core/server/master/src/main/java/alluxio/master/meta/UpdateCheck.java

```diff
@@ -79,11 +80,22 @@ public static String getLatestVersion(String clusterID, long connectionRequestTi
 
   /**
    * @param clusterID the cluster ID
-   * @return a string representation of the user's environment in the format "key1:value1, key2:
-   *         value2".
+   * @return a string representation of the user's environment in the format
+   *         "Alluxio/{ALLUXIO_VERSION} (valueA; valueB)"
    */
   @VisibleForTesting
   public static String getUserAgentString(String clusterID) throws IOException {
+    Joiner joiner = Joiner.on("; ").skipNulls();
+    String sysInfo = joiner.join(getUserAgentEnvironmentString(clusterID), getFeatureString());
```

Comment:
`getUserAgentEnvironmentString()` always returns a non-empty string but i believe we want `getFeatureString` to return null if there are zero features enabled. correct me if i'm wrong, but a joiner on no elements is going to return `""` and so you will end up joining `"some env string"` with `""` -> `"some env string; "` instead of `"some env string"`

regardless of the behavior of joiner, we should have a unit test for this case as i had asked for previously